### PR TITLE
8.0 mig purchase order line sequence

### PR DESCRIPTION
**Pull request by mathiasfrancke**
Hello,
I needed a module from Odoo 10 in Odoo 8. Now i would like to share the back port with the OCA.

Best Regards
Mathias

**Depends**
- [ ] [8.0 mig stock picking line sequence](https://github.com/OCA/stock-logistics-workflow#485)

### CODE REVIEW

File: purchase_order_line_sequence/README.rst
Comment (bodedra) [inline purchase_order_line_sequence/README.rst]: s/10.0/8.0

File: purchase_order_line_sequence/__openerp__.py
Comment (bodedra) [inline purchase_order_line_sequence/__openerp__.py]: Change website with `https://github.com/OCA/purchase-workflow`

File: purchase_order_line_sequence/models/purchase.py
Comment (bodedra) [inline purchase_order_line_sequence/models/purchase.py]: In version 8.0, these method available at `purchase.order` object with more arguments. Here is reference https://github.com/odoo/odoo/blob/8.0/addons/purchase/purchase.py#L840

Possible to add these method in test case?

**Review by bodedra — CHANGES_REQUESTED** [review]

**Comment by bodedra** [comment]
Additionally, These module is depend on `stock_picking_line_sequence` so first we have to back port it.

screenshot from 2018-12-18 22-39-02

**Comment by bodedra** [comment]
@mathiasfrancke I found your PR for that module at https://github.com/OCA/stock-logistics-workflow/pull/485

So can you update your PR description like

Depends:
-------------

- [ ] https://github.com/OCA/stock-logistics-workflow/pull/485

File: purchase_order_line_sequence/README.rst
Comment (mathiasfrancke) [reply purchase_order_line_sequence/README.rst]: corrected. comes with next push.

File: purchase_order_line_sequence/__openerp__.py
Comment (mathiasfrancke) [reply purchase_order_line_sequence/__openerp__.py]: corrected. comes with next push.

File: purchase_order_line_sequence/README.rst
Comment (nikul-serpentcs) [reply purchase_order_line_sequence/README.rst]: @mathiasfrancke @bodedra Use new README by fragment system, copying https://github.com/OCA/maintainer-tools/tree/master/template/module/readme and editing or removing the needed sections. You can then run locally oca-gen-addon-readme if you have OCA/maintainer-tools installed and you want to pregenerate the README.rst file (preferred), or keep previous README.rst file.

File: purchase_order_line_sequence/models/purchase.py
Comment (mathiasfrancke) [reply purchase_order_line_sequence/models/purchase.py]: Hello @bodedra ,
thanks for your note. I don't understand what you mentioned here. 
This is a method on the model `purchase.order.line`.
The method you linked is on the model `purchase.order`.
So both have the same name and different signatures but also the are on different objects.

Comment (bodedra) [reply purchase_order_line_sequence/models/purchase.py]: @mathiasfrancke In v8.0, we don't have same method in `purchase.order.line` compare to v10.0. But we have same method in `purchase.order` object level. 

Here is method references:

v8.0 - purchase.order - https://github.com/odoo/odoo/blob/8.0/addons/purchase/purchase.py#L840

v10.0 - purchase.order.line - https://github.com/odoo/odoo/blob/10.0/addons/purchase/models/purchase.py#L719

So we have make compatible code to work same in v8.0 & v.10.0

Comment (mathiasfrancke) [reply purchase_order_line_sequence/models/purchase.py]: Migrated missing method.

**Comment by mathiasfrancke** [comment]
Since there is a problem merging https://github.com/OCA/stock-logistics-workflow/pull/485 this PR is on hold from my side. It is requirement for this PR.

**Comment by mathiasfrancke** [comment]
I will close this PR because it is depenend on [stock_picking_line_sequence](https://github.com/OCA/stock-logistics-workflow/pull/485) where the PR is also closed because of duplicate with _stock_pciking_reorder_lines_ module

**Pull request closed without merging**
